main.rb

```diff
@@ -58,37 +62,44 @@ def my_select
     end
   end
 
-  def my_all?
+  def my_all?(opt = nil)
     bool_item = true
-    if block_given?
-      my_each do |item|
-        bool_item = false unless yield(item)
-      end
+    if opt.nil? && block_given?
+      my_each { |item| bool_item = false unless yield(item) }
+    elsif opt.is_a?(Regexp)
+      my_each { |item| bool_item = false if item.match(opt) }
+    elsif opt.is_a?(Module)
+      my_each { |item| bool_item = false unless item.is_a?(opt) }
+    else
+      my_each { |item| bool_item = false if item.nil? || item == false }
     end
     bool_item
   end
 
-  def my_any?
-    bool_item = true
-    if block_given?
-      bool_item = false
-      my_each do |item|
-        bool_item = true if yield(item)
-      end
+  def my_any?(opt = nil)
+    bool_item = false
+    if opt.nil? && block_given?
+      my_each { |item| bool_item = true if yield(item) }
+    elsif opt.is_a?(Regexp)
+      my_each { |item| bool_item = true if item.match(opt) }
+    elsif opt.is_a?(Module)
+      my_each { |item| bool_item = true if item.is_a?(opt) }
+    else
+      my_each { |item| bool_item = true unless item.nil? || item == false }
     end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`any?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-any-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_any.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] when a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
```ruby
words.my_any?('cat') == words.any?('cat')
```
Should return `true`